framework/src/st_things/things_stack/framework/things_data_manager.c

```diff
@@ -1133,6 +1179,14 @@ static int parse_things_info_json(const char *filename)
 					node->sig_cnt = cJSON_GetArraySize(single);
 					THINGS_LOG_D(THINGS_DEBUG, TAG, "[DEVICE] Resources for Single Usage Cnt : %d", node->sig_cnt);
 
+					// Add 2 single resources for to support accesslist, provisioning.
+					/***** additional resource(provisininginfo / accesspointlist) *****/
+					int resCnt = sizeof(gstResources) / sizeof(things_resource_info_s);
+					node->single = (things_resource_info_s *)things_malloc(sizeof(things_resource_info_s) * (node->sig_cnt + resCnt));
+					if(node->single == NULL) {
+						return 0;
```

Comment:
Almost APIs use 0 as OK and -1 as ERROR. We also have definitions OK to 0 and ERROR to -1.
But this API uses 0 as failure. Is it ok? How about use -1 instead of 0?